sql/core/src/test/resources/sql-tests/results/pipe-operators.sql.out

```diff
@@ -650,6 +650,22 @@ org.apache.spark.sql.AnalysisException
 }
 
 
+-- !query
+table t
+|> where first_value(x) over w = 1
+   window w as (partition by y)
+-- !query schema
+struct<>
+-- !query output
+org.apache.spark.sql.AnalysisException
+{
+  "errorClass" : "_LEGACY_ERROR_TEMP_1034",
```

Comment:
do you think we could also add `windowClause?` to the `whereClause` in the `operatorPipeRightSide`, just for purposes of throwing a better error message in this case? It could look something like:

```
operatorPipeRightSide
    : selectClause windowClause?
    // Note that the WINDOW clause is not allowed in the WHERE pipe operator, but we add it here in
    // the grammar simply for purposes of catching this invalid syntax and throwing a specific
    // dedicated error message.
    | whereClause windowClause?
```

and in the AstBuilder:

```
    }.getOrElse(Option(ctx.whereClause).map { c =>
      if (ctx.windowClause != nullptr) {
        throw QueryParsingErrors.windowClauseInOperatorPipeWhereClauseNotAllowedError(ctx)
      }
      withWhereClause(c, withSubqueryAlias())
```